src/App/Utilities/VerificationActionsFlowHelper.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Bit.App.Abstractions;
+using Bit.App.Pages;
+using Bit.App.Pages.Accounts;
+using Bit.Core.Abstractions;
+using Bit.Core.Enums;
+using Bit.Core.Utilities;
+using Xamarin.Forms;
+
+namespace Bit.App.Utilities
+{
+    public interface IVerificationActionsFlowHelper
+    {
+        IVerificationActionsFlowHelper Configure(VerificationFlowAction action,
+            IActionFlowParmeters parameters = null,
+            string verificatioCodeMainActionText = null,
+            bool isVerificationCodeMainActionStyleDanger = false);
+        IActionFlowParmeters GetParameters();
+        Task ValidateAndExecuteAsync();
+        Task ExecuteAsync(IActionFlowParmeters parameters);
+    }
+
+    public interface IActionFlowParmeters
+    {
+        string Secret { get; set; }
+    }
+
+    public class DefaultActionFlowParameters : IActionFlowParmeters
+    {
+        public string Secret { get; set; }
+    }
+
+    public interface IActionFlowExecutioner
+    {
+        Task Execute(IActionFlowParmeters parameters);
+    }
+
+    public enum VerificationFlowAction
+    {
+        ExportVault,
+        DeleteAccount
+    }
+
+    /// <summary>
+    /// Verifies and execute actions on the corresponding order.
+    /// 
+    /// Use: From the caller
+    /// - Configure it on <see cref="Configure(VerificationFlowAction, IActionFlowParmeters)"/>
+    /// - Call <see cref="ValidateAndExecuteAsync"/>
+    /// 
+    /// For every <see cref="VerificationFlowAction"/> we need an implementation of <see cref="IActionFlowExecutioner"/>
+    /// and it to be configured in the inner dictionary.
+    /// Also, inherit from <see cref="DefaultActionFlowParameters"/> if more custom parameters are needed for the executioner.
+    /// </summary>
+    public class VerificationActionsFlowHelper : IVerificationActionsFlowHelper
+    {
+        private readonly IKeyConnectorService _keyConnectorService;
+        private readonly IPasswordRepromptService _passwordRepromptService;
+
+        private VerificationFlowAction? _action;
+        private IActionFlowParmeters _parameters;
+        private string _verificatioCodeMainActionText;
+        private bool _isVerificationCodeMainActionStyleDanger;
+
+        private readonly Dictionary<VerificationFlowAction, IActionFlowExecutioner> _actionExecutionerDictionary = new Dictionary<VerificationFlowAction, IActionFlowExecutioner>();
+
+        public VerificationActionsFlowHelper(IKeyConnectorService keyConnectorService,
+            IPasswordRepromptService passwordRepromptService)
+        {
+            _keyConnectorService = keyConnectorService;
+            _passwordRepromptService = passwordRepromptService;
+
+            _actionExecutionerDictionary.Add(VerificationFlowAction.DeleteAccount, ServiceContainer.Resolve<IDeleteAccountActionFlowExecutioner>("deleteAccountActionFlowExecutioner"));
```

Comment:
I like what you're doing with the `VerificationActionsFlowHelper` and the `IActionFlowExecutioner`, trying to keep the two steps decoupled. I guess I'm wondering what you see are some other use cases for this workflow? This might just help me understand it better.

If the goal _is_ to keep it decoupled from the ViewModel and allow us to use it during export as well, I'm not sure if registering the `IDeleteAccountActionFlowExecutioner` as a service and keeping a reference to it here is the best way to tackle it. Do you think we need access to the Executioner outside of the Helper? Could we just create a new object inside the helper?

Reply:
Yes, the other current use case would be "Export Vault" and any other future action that need verification ✅ .
I thought it as this will be the helper needed for any interaction with the master password/verification code/any future verification process to perform an action.

About the decoupling, I think it's better to have it registered as a service in order to be more testable (I forgot to mention that I'd need to add unit tests on the PR), so we can just inject the `Executioner` from the testing container. Maybe we could improve that and just pass it on the constructor so we avoid resolving it there. Or put a public method to setup the executioners from outside.
Also I wasn't sure on having the `Executioner` on the same `ViewModel` file or in another file in some specific folder or `Utilities`.

Reply:
Ah I see, that sounds good to me. After thinking about it I think including the `Executioner` in the `VerificationActionsFlowHelper` as a service is a better idea than exposing it and leaving it in the constructor or something. 

I'm also ok with leaving the `Executioner` in the `ViewModel`. Makes more sense to me than throwing it in a random folder somewhere.